src/conventions/h-group/take-action.js

```diff
@@ -263,10 +259,22 @@ export function take_action(game) {
 		}
 	}
 
+	// Consider finesses while finessed if we are only waited on to play one card,
+	// it's not a selfish finesse, doesn't require more than one play from our own hand,
+	// and we're not in the end-game.
+	const waiting_self_connections = game.common.waiting_connections.filter(c => c.connections[0]?.reacting === state.ourPlayerIndex).length;
+	const is_selfish = best_play_clue?.result.playables[0]?.playerIndex === state.ourPlayerIndex;
```

Comment:
This isn't quite the condition for a *Selfish Finesse*. What I had in mind was a finesse where we are holding a card needed to start, like holding known r1 and cluing r3 to finesse r2. Around line 359 of take-action.js I implemented a basic selfish check for 2p, though I think it should be the same for more players.

Reply:
I was mistakenly assuming that playables was in order of play. It looks like not_selfish on line 359 will consider it selfish if any of the plays are in our hand, however, say that we are holding known r2, it shouldn't be selfish to clue the r3 finesse.